MagicalRecord/Categories/NSManagedObject/NSManagedObject+MagicalRecord.h

```diff
@@ -66,6 +66,12 @@
 
 @end
 
+@protocol MagicalRecord_MergeSkippable <NSObject>
+
+- (BOOL) MR_maySkipMerging;
```

Comment:
As far as I understand, when it returns `true`, it tries to skip merging in a conformed object.

If so, could it be `MR_shouldSkipMerging`? Using `may` can be ambiguous regarding the result and whether skipping or not depends on the the implementation side in the conformed object.

Reply:
You are right. The way I have updated MagicalRecord code now is that it obeys the value of this property. Initially, I thought of adding a delegate to MagicalRecord that handles the filtering, and then it would be up to the delegate to decide if we should skip the merge or not (then the current naming would make sense - an object indicates that the merge may be skipped but whether it is skipped or not depends on whether the delegate is set and if it actually skips).
I dropped the delegate idea though because there is no instance on which to set the delegate - MagicalRecord relies heavily on static variables, so it would feel kind of messy. Also - adding the delegate would still require me to change `rootContextDidSave` significantly, so I decided to keep it simple.